benches/server.rs

```diff
@@ -7,33 +7,61 @@ extern crate pretty_env_logger;
 extern crate test;
 
 use std::io::{Read, Write};
-use std::net::{TcpListener, TcpStream, SocketAddr};
+use std::net::{TcpListener, TcpStream};
 use std::sync::mpsc;
 
 use futures::Future;
 use futures::sync::oneshot;
 
-use hyper::header::{ContentLength, ContentType};
+use hyper::header::{ContentLength, ContentType, TransferEncoding};
 use hyper::server::{self, Service};
 
+macro_rules! bench_server {
+    ($b:ident, $handler:ident, $total_bytes:expr) => {
+        let (_until_tx, until_rx) = oneshot::channel();
+        let addr = {
+            let (addr_tx, addr_rx) = mpsc::channel();
+            ::std::thread::spawn(move || {
+                let addr = "127.0.0.1:0".parse().unwrap();
+                let srv = hyper::server::Http::new().bind(&addr, || Ok($handler)).unwrap();
+                let addr = srv.local_addr().unwrap();
+                addr_tx.send(addr).unwrap();
+                srv.run_until(until_rx.map_err(|_| ())).unwrap();
+            });
+
+            addr_rx.recv().unwrap()
+        };
+
+        let mut tcp = TcpStream::connect(addr).unwrap();
+        let mut buf = [0u8; 4096];
 
+        $b.bytes = 130 + 35;
```

Comment:
This should include the total bytes part. This is what let's the benchmark show the MB/s.